sdks/python/apache_beam/transforms/external.py

```diff
@@ -56,13 +57,12 @@ class ExternalTransform(ptransform.PTransform):
   _EXPANDED_TRANSFORM_UNIQUE_NAME = 'root'
   _IMPULSE_PREFIX = 'impulse'
 
-  def __init__(self, urn, payload, endpoint):
-    if grpc is None and isinstance(endpoint, str):
+  def __init__(self, urn, payload, expansion_service):
+    if grpc is None and isinstance(expansion_service, str):
```

Comment:
We should add a docstring to `__init__`.  It's unclear what type `expansion_service` should be if it's not a string. 

I can see that this PR makes that a bit more complicated, as the new `JavaJarExpansionService` does not inherit from `beam_expansion_api_pb2_grpc.ExpansionServiceServicer`, but we can list the currently valid options (`expansion_service.ExpansionServiceServicer` and `JavaJarExpansionService`) and/or describe the protocol ("must have an Expand method that takes...").  

Note that when we add support for typing we can use `Protocols` to describe this in more concrete terms.

Reply:
Good point; added a docstring. Yes, this supports anything that implements this service (a servicer, a stub, or anything else that does so by virtual of an Expand method).